src/lib/y2packager/product_location.rb

```diff
@@ -87,5 +87,12 @@ def summary
 
       details.summary
     end
+
+    # Product dependencies taken from its details
+    #
+    # @return [Array] empty list if there is no dependencies or the dependencies list
+    def depends_on
+      details&.depends_on || []
```

Comment:
Hm, I'd rather fix this line https://github.com/yast/yast-packager/blob/master/src/lib/y2packager/dialogs/addon_selector.rb#L222 instead adding a wrapper here. In the end we would add wrappers for all methods and that would defeat the purpose of having a separate `ProductLocationDetails` class...

Reply:
That's was a doubt that prompt to my mind while I was solving the issue. And also I think that the `ProductLocationDetails#initialize` should ensure to _sanitize_ the `depends_on` param, because is expected that it will be always an array, isn't it? (see https://github.com/yast/yast-packager/blob/79d45133b84d6afa0de9afe95e816dfd66d85731/src/lib/y2packager/dialogs/addon_selector.rb#L152-L156)

Reply:
The `nil` value for the `depends_on` is used to indicate that the were dependency issues and the dependencies could not have been computed.

Currently that's is not used in the UI, but in theory we could display some error/warning message instead of an empty list.

Reply:
The problem I see with the `ProductLocation` and `ProductLocationDetails` approach is that the details attribute could be nil. This happens because `:product` is an optional parameter in `ProductLocaltion#initialize`.

So, checking the existence of details is always mandatory: `product.details&.attribute`. And this has an implicit drawback (IMHO). The caller is the responsible to define the default value when details or the attribute are nil:

```
product.details&.attribute || []
```

For me, a better solution would be to force `ProductLocation` to always have a details object. And then I would define proper default values for each detail attribute:

```
class ProductLocation
  def initialize(name, dir, product: nil)
    @details = product || ProductLocationDetails.new
  end
end

class ProductLocaltionDetails
  def initialize(product: nil, depends_on: [], ...)
  end
end
```

Reply:
>  And then I would define proper default values for each detail attribute

But in the previous comment, @lslezak said

>  The nil value for the depends_on is used to indicate that the were dependency issues and the dependencies could not have been computed.And then I would define proper default values 

But I still thinking that this will be problematic, since

```irb
irb(main):001:0> [].concat(nil)
Traceback (most recent call last):
        5: from /usr/bin/irb:23:in `<main>'
        4: from /usr/bin/irb:23:in `load'
        3: from /usr/lib64/ruby/gems/2.6.0/gems/irb-1.0.0/exe/irb:11:in `<top (required)>'
        2: from (irb):1
        1: from (irb):1:in `concat'
TypeError (no implicit conversion of nil into Array)
irb(main):002:0> 
```

Reply:
Yes, I don't know all the design decisions behind this API. But I would use another mechanism to indicate dependency issues. I would keep `#depends_on` as an empty list even when there are issues with dependencies (this avoid to check for nil everywhere), and also I would extend the API with an specific method to ask for possible issues: `#dependency_errors?`.  But, again, this is only my personal feeling, Ladislav knows the details much better.

Reply:
> For me, a better solution would be to force `ProductLocation` to always have a details object.

In some cases the repository might not contain a product (e.g. a custom or a 3rd party repository like Packman). In that case we do not have any details, we can only read the data from the `/media.1/products` file which only contains directory name and add-on name.

In that case it would make sense to have `nil` details

Reply:
@lslezak, @joseivanlopez, 

All things considered, I'll fix the issue in https://github.com/yast/yast-packager/blob/master/src/lib/y2packager/dialogs/addon_selector.rb#L222, although I have a similar feeling than Iván.

Reply:
OK, I'd prefer a quick/simple fix to avoid crashes, we can discuss and improve it later...